### DISCUSSION TITLE
Errors and inconsistencies with new "in-house" Linux installation

### DISCUSSION THREAD

**Discussion by RichardHaselgrove** [BOINC/boinc#5781] [General]
I have today taken delivery of a brand-new PC. It was supplied without an OS, and I have installed Linux Mint 22, based on Ubuntu 24.04

So I followed the instructions from https://boinc.berkeley.edu/linux_install.php, for Ubuntu 24.04. It offers three versions: Stable (v8.0.2), Alpha (v8.0.4), and Nightly (v8.1.0)

I tried the first two. Both installed v7.24.1, and said that this was the latest available.

BOINC was installed in user mode. I opened the Manager by typing 'boincmgr' in terminal: no launch icon was created in the GUI, and I haven't found a way of creating one yet.

The version of the Manager launched is the one which was hacked about by the Debian distro maintainers: it is designed to match systemd installations in corporate environments, where employees are not allowed to stop or manage the BOINC client themselves.

- there is no distinction between 'close Window' and 'exit Manager'. Close is the same as exit.
- All the other choices are available - Run the client?, Enable Manager exit dialog?, Enable Client shutdown dialog? But none of them work.
- I can find no way of requesting an orderly shutdown of the client, other than powering down the machine.

This seems to be a half-baked mish-mash of differing objectives. I think it ought to be labelled 'Work in Progress', at the very least, rather than presented as 'the way to do it'.

**Comment by AenBleidd** [comment]
> I tried the first two. Both installed v7.24.1, and said that this was the latest available.

This is definitely wrong and should not happen.
Something definitely was wrong (either made by you or wrong in the instruction).

**Comment by RichardHaselgrove** [comment]
Well, I copied and pasted the command lines into a terminal window:

> sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 40254C9B29853EA6
sudo apt-add-repository deb https://boinc.berkeley.edu/dl/linux/stable/noble noble main
sudo apt install boinc-client boinc-manager

**Reply by AenBleidd** [reply]
Have you tried to do
sudo apt update
before doing
sudo apt install ...
?

**Reply by RichardHaselgrove** [reply]
No, but I have now.

`All packages are up-to-date.
boinc-client is already the newest version (7.24.1+dfsg-4build1).
boinc-manager is already the newest version (7.24.1+dfsg-4build1).
0 to upgrade, 0 to newly install, 0 to remove and 0 not to upgrade.
`

**Reply by kiska3** [reply]
> Have you tried to do sudo apt update before doing sudo apt install ... ?

apt update is definitely required, but to a normal user apt update isn't something they run, just see the LTT Linux challenge for that.

Normal users copy commands and assume everything works as intended.

**Reply by AenBleidd** [reply]
Hm, that's odd. Will check it tonight

**Reply by AenBleidd** [reply]
> > Have you tried to do sudo apt update before doing sudo apt install ... ?
> 
> 
> 
> apt update is definitely required, but to a normal user apt update isn't something they run, just see the LTT Linux challenge for that.
> 
> 
> 
> Normal users copy commands and assume everything works as intended.

Of course, and that's why I'll double check that.

**Comment by RichardHaselgrove** [comment]
This may add more data:

`sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 40254C9B29853EA6
[sudo] password for richard:         
Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8)).
Executing: /tmp/apt-key-gpghome.GNVfvj0gP6/gpg.1.sh --keyserver hkp://keyserver.ubuntu.com:80 --recv 40254C9B29853EA6
gpg: key 40254C9B29853EA6: "Vitalii Koshura (BOINC DEB and RPM repos key) <lestat.de.lionkur@gmail.com>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
`

Also, I see that lower down, the Ubuntu instructions say:

GPU computing

If the system has an NVIDIA GPU, you can enable CUDA apps using

`sudo apt-get install nvidia-cuda-toolkit`

The CUDA toolkit is a developer resource, used by GPU application programmers. It's not needed by end users: the necessary runtime support (for both CUDA and OpenCL) is included with the GPU drivers downloaded from NVidia.

**Comment by davidpanderson** [comment]
Are there any distro/GPU combinations where users need to install something
in order to run CUDA or OpenCL apps?

**Comment by RichardHaselgrove** [comment]
Today's installation detected that I had a GPU which needed a driver. It offered a choice: preselected was a 'nouveau' open-source driver, but the official NVidia driver was 'recommended'.

In general, I think the BOINC community has decided that the open-source drivers are to be avoided, but the OEM drivers are OK. I can't speak for AMD / ATI cards.

**Comment by davidpanderson** [comment]
I changed the Linux install page to offer more generic advice about GPU drivers.

**Reply by RichardHaselgrove** [reply]
Thanks - I think that's better. From what I remember from early testing at SETI Beta, the toolkit contained debug versions of the drivers, which were not licensed for public release.

**Comment by AenBleidd** [comment]
Hm, looks like for some reason 'sudo apt-add-repository' doesn't work on linux.
You can add BOINC linux repository using the 'Software Sources' tool:
![image](https://github.com/user-attachments/assets/eeac7659-0449-4257-9c9e-81f339e26b2f)

**Reply by RichardHaselgrove** [reply]
Got there eventually, after a bit of a struggle - Mint sometimes makes you switch between command line and GUI, and it's not always easy to translate. Machine is now running first test tasks under v8.0.4 Alpha.

The switch of data location from boinc-client to boinc has caught me out - it's not using the right data drive, and it hasn't picked up my pre-prepared configuration files. But I can sort that out later.

I've also spotted a couple of problems, which I'll report as alpha issues when I've had a chance to test them more carefully.
